packages/block-library/src/heading/edit.js

```diff
@@ -12,49 +12,37 @@ import HeadingToolbar from './heading-toolbar';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { PanelBody } from '@wordpress/components';
-import { compose } from '@wordpress/compose';
+import { PanelBody, withFallbackStyles } from '@wordpress/components';
 import { createBlock } from '@wordpress/blocks';
 import {
 	AlignmentToolbar,
 	BlockControls,
 	InspectorControls,
 	RichText,
-	withColors,
-	PanelColorSettings,
+	__experimentalUseColors,
 } from '@wordpress/block-editor';
-import { memo } from '@wordpress/element';
 
-const HeadingColorUI = memo(
-	function( {
-		textColorValue,
-		setTextColor,
-	} ) {
-		return (
-			<PanelColorSettings
-				title={ __( 'Color Settings' ) }
-				initialOpen={ false }
-				colorSettings={ [
-					{
-						value: textColorValue,
-						onChange: setTextColor,
-						label: __( 'Text Color' ),
-					},
-				] }
-			/>
-		);
-	}
-);
+/**
+ * Browser dependencies
+ */
+const { getComputedStyle } = window;
 
 function HeadingEdit( {
+	backgroundColor,
 	attributes,
 	setAttributes,
 	mergeBlocks,
 	onReplace,
 	className,
-	textColor,
-	setTextColor,
 } ) {
+	const { TextColor, InspectorControlsColorPanel } = __experimentalUseColors(
```

Comment:
Good work, let's try to open follow-up PRs to use in other blocks to see how it behaves, if it's good enough.

Reply:
What does `<TextColor>` do? This doesn't seem to be documented anywhere.

Reply:
It applies the text color as inline style to the wrapped component.

Reply:
Why not pass the style directly to the element?

Reply:
It's less verbose this way.

Reply:
I'm on the fence personally, if it's just about "style", I guess being explicit is better (passing styles). But I believe it also handles classNames and merging these props which ultiimately can result in a lot of duplicated code.

Reply:
It looks a bit like magic right now. It's not very obvious what it does. I'll look around sometime to see how it works.

Reply:
>But I believe it also handles classNames and merging these props which ultiimately can result in a lot of duplicated code.

Yes